public/app/features/alerting/unified/components/rule-viewer/tabs/AlertVersionHistory.tsx

```diff
@@ -54,6 +57,11 @@ export function AlertVersionHistory({ ruleUid }: AlertVersionHistoryProps) {
   const [checkedVersions, setCheckedVersions] = useState(new Set<string>());
   const canCompare = useMemo(() => checkedVersions.size > 1, [checkedVersions]);
 
+  // check if restoring is allowed/enabled
+  const [restoreSupported, restoreAllowed] = useAlertRuleAbility(combinedRule, AlertRuleAction.Restore);
```

Comment:
There is a new `useRulerRuleAbility` hook, which doesn't need a combined rule, but ruler rule and group identifier.
I'd highly encourage you to use that instead, so we don't create new code which rely on combined rules